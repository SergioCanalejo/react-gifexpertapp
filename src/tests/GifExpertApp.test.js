const { shallow } = require('enzyme');
const { default: GifExperApp } = require('../GifExpertApp');
const React = require('react');

describe('Testing GifExpertApp', () => {
  const defaultCategories = ['One Piece'];

  afterEach(() => {
    jest.clearAllMocks();
  });
  test('Must show the app correctly', () => {
    const wrapper = shallow(<GifExperApp defaultCategories={defaultCategories} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should have two GifGrids', () => {
    const multipleCategories = ['Dragon Ball', ...defaultCategories];
    const wrapper = shallow(<GifExperApp defaultCategories={multipleCategories} />);
    expect(wrapper.find('GifGrid').length).toBe(multipleCategories.length);
  });
});
