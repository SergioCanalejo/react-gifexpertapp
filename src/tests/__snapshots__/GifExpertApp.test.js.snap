// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp Must show the app correctly 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Piece"
    key="cat-One Piece"
  />
</Fragment>
`;
